packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2274,6 +2274,24 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
                 child: showHint ? hintTextWidget : const SizedBox.shrink(),
               );
     }
+    if(decoration.hint != null) {
+      final Widget? hintTextWidget = decoration.hint;
+      final bool showHint = isEmpty && !_hasInlineLabel;
+       hint =
+          maintainHintHeight
+              ? AnimatedOpacity(
+                opacity: showHint ? 1.0 : 0.0,
+                duration: decoration.hintFadeDuration ?? _kHintFadeTransitionDuration,
+                curve: _kTransitionCurve,
+                child: hintTextWidget,
+              )
+              : AnimatedSwitcher(
+                duration: decoration.hintFadeDuration ?? _kHintFadeTransitionDuration,
+                transitionBuilder: _buildTransition,
+                child: showHint ? hintTextWidget : const SizedBox.shrink(),
+              );
+    }
+
```

Comment:
Instead of duplicating this logic, consider to initialize `hintTextWidget` such as `final Text hintTextWidget = decoration.hint ?? Text(`

Reply:
optimized the code